zebra/zserv.c

```diff
@@ -934,6 +934,71 @@ zserv_rnh_unregister (struct zserv *client, int sock, u_short length,
   return 0;
 }
 
+/* FEC register */
+static int
+zserv_fec_register (struct zserv *client, int sock, u_short length)
+{
+  struct stream *s;
+  struct zebra_vrf *zvrf;
+  u_short l = 0;
+  struct prefix p;
+  u_int16_t flags;
+  u_int32_t label_index = MPLS_INVALID_LABEL_INDEX;
+
+  s = client->ibuf;
+  zvrf = vrf_info_lookup(VRF_DEFAULT);
+  if (!zvrf)
+    return 0; // unexpected
+
+  while (l < length)
+    {
+      flags = stream_getw(s);
```

Comment:
I'd like to have some sanity checks here.

zebra, being a single point of failure, should be solid as a rock. A bug in bgpd or in any other client daemon should not cause it to crash.